core/controllers/android_e2e_config.py

```diff
@@ -79,11 +81,13 @@ def post(self):
 
         if not constants.DEV_MODE:
             raise Exception('Cannot load new structures data in production.')
-        if topic_services.does_topic_with_name_exist(
-                'Android test'):
-            topic = topic_fetchers.get_topic_by_name('Android test')
+        if topic_services.does_topic_with_name_exist('Android test'):
+            topic = topic_fetchers.get_topic_by_name(
+                'Android test', strict=True
+            )
             topic_rights = topic_fetchers.get_topic_rights(
-                topic.id, strict=False)
+                topic.id, strict=True
```

Comment:
This is a functional change from before--why is it necessary?

Reply:
If we keep `strict=False` then `topic_rights` can be a None value and if we see line 93 just below this line, there we are not accounting for the case that topic_rights can be None and accessing the `topic_is_published` attribute. So, if topic_rights is None, then a python error will occur because there is no `topic_is_published` attribute on None. So, just to throw a more explicit error than the python attribute error on None. I used `strict=true` here. Also, `strict=true` also narrows down the type, now topic_rights can only be a non-None value.

Reply:
Got it--thanks for the explanation!